examples/aggregates/README.md

```diff
@@ -0,0 +1,60 @@
+- When new aggregated object values are published on a server owned interface, then the device will
+publish new aggregated object values on the device owned interface.
+
+If you want to start using a local instance of Astarte hosted on your machine and register the
+device automatically you can check out the [`toggle_led` example](../toggle_led) in this repository.
+
+We will assume you already performed the common configuration and satisfy the common prerequisites.
+
+## Device connection to Astarte
+
+Once the example has been flashed it will automatically attempt to connect the device to the remote
+instance of Astarte.
+Verify this by using the monitor functionality of the ESP-IDF to observe the log messages of the
+ESP32.
+After connection has been accomplished, the following ESP32 log message should be shown:
+```
+ASTARTE_EXAMPLE_AGGREGATE: Astarte device connected, session_present: 0
+```
+You can see how the connection has been achieved also from the Astarte Dashboard by opening the
+newly created device and observing the status that should have changed to *Connected*.
+
+## Send and receive data from the device
+
+Now that our device is connected to Astarte is time to start publishing data.
+We can interact with the remote Astarte instance using the `astartectl` tool.
+
+We will use two commands in this example, `send-data` and `get-samples`.
+With `send-data` we can publish new values for a server owned interface.
+The syntax is the following:
+```
+astartectl --astarte-url "<API_URL>" --token "<TOKEN>" appengine --realm-name "<REALM_NAME>" \
+    devices send-data <DEVICE_NAME>                                                          \
+    org.astarteplatform.esp32.examples.ServerAggregate /11                                   \
+    '{"longinteger_endpoint":45993543534, "booleanarray_endpoint":[false,false,true,true]}'
+```
+Where `<TOKEN>` is the shell variable containing the token, `<REALM_NAME>` is your realm's name and
+`<API_URL>` is the API's URL as found on astarte cloud console. `<DEVICE_NAME>` is the device ID to
+send the data to. The last argument is the data to publish.
+
+The device will react to this event by publishing some values on its interface. The value
+published in this example will correspond to the negation of the received value.
+
+We can check if the publish from the device has been correctly received on the server using
+`get-samples` as follows:
+```
+astartectl --astarte-url "<API_URL>" --token "<TOKEN>" appengine --realm-name "<REALM_NAME>" \
+    devices get-samples <DEVICE_NAME>                                                        \
+    org.astarteplatform.esp32.examples.DeviceAggregate /24 -c 1
+```
+This will print the latest published data from the device that has been stored in the Astarte Cloud
+instance.
+To print a longer history of the published data change the number in `-c 1` to the wanted history
```

Comment:
"to the _preferred_/_desired_ history" or rephrase to "to the history length of your liking."